pkg/apis/core/v1/conversion_test.go

```diff
@@ -346,3 +346,76 @@ func roundTripRS(t *testing.T, rs *apps.ReplicaSet) *apps.ReplicaSet {
 	}
 	return obj3
 }
+
+func Test_core_PodStatus_to_v1_PodStatus(t *testing.T) {
+	// core to v1
+	var v1PodStatus v1.PodStatus
+	corePodStatus := core.PodStatus{
+		PodIPs: []core.PodIPInfo{
+			{
+				IP:         "1.1.1.1",
+				Properties: make(map[string]string),
+			},
+			{
+				IP:         "2.2.2.2",
+				Properties: make(map[string]string),
+			},
+		},
+	}
+	// convert
+	if err := corev1.Convert_core_PodStatus_To_v1_PodStatus(&corePodStatus, &v1PodStatus, nil); nil != err {
+		t.Errorf("Convert core.PodStatus to v1.PodStatus failed with error %v", err.Error())
+	}
+
+	// validate results
+	if v1PodStatus.PodIP != corePodStatus.PodIPs[0].IP {
+		t.Errorf("Convert core.PodStatus to v1.PodStatus failed. Expected v1.PodStatus=%v but found %v", corePodStatus.PodIPs[0].IP, v1PodStatus.PodIP)
+	}
+
+	// match v1.PodIPs to core.PodIPs
+	for idx, _ := range corePodStatus.PodIPs {
+		if v1PodStatus.PodIPs[idx].IP != corePodStatus.PodIPs[idx].IP {
+			t.Errorf("Convert core.PodStatus to v1.PodStatus failed. Expected v1.PodStatus[%v]=%v but found %v", idx, corePodStatus.PodIPs[idx].IP, v1PodStatus.PodIPs[idx].IP)
+		}
+	}
+}
+func Test_v1_PodStatus_to_core_PodStatus(t *testing.T) {
+	var corePodStatus core.PodStatus
+	// Error case PodIP != PodIPs[0]
+	v1PodStatus := v1.PodStatus{
+		PodIP: "0.0.0.0",
+		PodIPs: []v1.PodIPInfo{
+			{IP: "1.1.1.1"},
+			{IP: "2.2.2.2"},
+		},
+	}
+	// should return err
+	if err := corev1.Convert_v1_PodStatus_To_core_PodStatus(&v1PodStatus, &corePodStatus, nil); err == nil {
+		t.Errorf("Convert v1.PodStatus to core.PodStatus failed. Expected an error when corePodStatus.PodIP != corePodStatus.PodIPs[0].IP")
+	}
+
+	// reset
+	corePodStatus = core.PodStatus{}
+	v1PodStatus.PodIP = "" // only list of ip provided
+	if err := corev1.Convert_v1_PodStatus_To_core_PodStatus(&v1PodStatus, &corePodStatus, nil); err != nil {
+		t.Errorf("Convert v1.PodStatus to core.PodStatus failed with error:%v", err.Error())
+	}
+	// match ip list
+	for idx, _ := range v1PodStatus.PodIPs {
+		if corePodStatus.PodIPs[idx].IP != v1PodStatus.PodIPs[idx].IP {
+			t.Errorf("Convert v1.PodStatus to core.PodStatus failed core.PodIPs[%v]=%v expected %v", idx, corePodStatus.PodIPs[idx].IP, v1PodStatus.PodIPs[idx].IP)
+		}
+	}
+
+	// reset
+	corePodStatus = core.PodStatus{}
+	v1PodStatus.PodIP = "1.1.1.1" //only PodIP Provided
```

Comment:
missing the case of podIP == podIPs[0] ?

Reply:
083bf4e adds it.